models/meshmodel/registry/host.go

```diff
@@ -34,8 +59,8 @@ func (h *Host) AfterFind(tx *gorm.DB) error {
 		h.IHost = ArtifactHub{}
 	case "kubernetes":
 		h.IHost = Kubernetes{}
-	default:
-		return ErrUnknownHost(errors.New("unable to find compatible host for the component"))
+	default: // do nothing if the host is not pre-unknown. Currently adapters fall into this case.
```

Comment:
This change relates to the question I asked here: https://github.com/meshery/meshkit/pull/375#issuecomment-1738593411